src/wamr/WAMRWasmModule.cpp

```diff
@@ -187,29 +199,139 @@ int WAMRWasmModule::executeWasmFunction(const std::string& funcName)
     } else {
         std::string errorMessage(
           ((AOTModuleInstance*)moduleInstance)->cur_exception);
-        SPDLOG_ERROR("Function failed: {}", errorMessage);
+        SPDLOG_ERROR("Caught wasm runtime exception: {}", errorMessage);
+
+        // TODO - agree on a protocol to send the return code through the error
+        // message (throwing exceptions not supported). For the moment, if the
+        // error message is a "Exception: 0" we assume no errors (i.e. exit(0))
+        if (errorMessage == "Exception: 0") {
+            return 0;
+        }
 
         return 1;
     }
 
     return 0;
 }
 
+void WAMRWasmModule::writeStringToWasmMemory(const std::string& strHost,
+                                             char* strWasm)
+{
+    this->validateNativeAddress(strWasm, strHost.size());
```

Comment:
We don't use `this->` unless it's absolutely necessary to avoid collisions between function names (and even then we should reconsider renaming the function).